ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGen/policyBuilder.go

```diff
@@ -113,11 +109,17 @@ func (pbuilder *PolicyBuilder) getCustomResource(data map[string]interface{},spe
 	// Get resource name from source policy if name is empty or N/A
 	if name == "" || name == utils.NotApplicable {
 		name = sourcePolicyMap["metadata"].(map[string]interface{})["name"].(string)
+	} else {
+		sourcePolicyMap["metadata"].(map[string]interface{})["name"] = name
+	}
+
+	if len(labels) != 0 {
```

Comment:
usually labels are not changing on the go, its either defined on the CR and it will be used with every generated policy. OR u just need to add them for specific clusters. like its not a config to change its more a key field that will be used to match a selection later with other CR.

Reply:
Ok. Just wanted to check if deleting all labels would be needed.